src/components/Projects/ProjectItem/index.jsx

```diff
@@ -11,18 +16,68 @@ const ProjectItem = ({
   description,
   attribution,
   projectUrl,
+  detailMdName,
   githubUrl,
+  useDetailMd,
 }) => {
+  const { isModalShown, showModal, hideModal } = useModal();
+  const [modalContent, setModalContent] = useState("");
   const addDefaultSrc=({target}) => (target.src = noImage)
 
+  const ProjectDetailWithLink = ({ children }) => 
+    <a
+      className='project__details'
+      href={projectUrl}
+      target='_blank'
+      rel='noopener noreferrer'
+    >{children}</a>
+
+  const ProjectDetailWithModal = ({ children }) => {
+    return (
+      <>
+      <button className='project__details' onClick={showModal}>
+        {children}  
+      </button>
+      {isModalShown && 
+        <ModalPortal onClose={hideModal}>
+          <div className="markdown">
+            {parse(modalContent)}
+          </div>
+        </ModalPortal>
+      }
+      </>
+    )
+  }
+
+  const ProjectDetail = ({ children }) => 
+    useDetailMd 
+      ? <ProjectDetailWithModal children={children} />
+      : <ProjectDetailWithLink children={children} />
+
+  useEffect(() => {
+    if (!detailMdName) return;
+
+    import(`../../../assets/static/markdown/${detailMdName}`)
+      .then((res) => {
+        fetch(res.default)
+          .then((res) => res.text())
+          .then((res) => setModalContent(mmd(res)))
+          .catch((err) => console.log(err));
+      })
+      .catch((err) => {
+        console.log(err);
+
+        return '';
+      });
```

Comment:
`import`를 하실 때, 사용자가 굳이 파일명을 `db.json` 에 언급하지 않아도 되도록 
아래와 같이 변경하는 것은 어떠신가요?
혹시 다른 이유가 있으셨던 것이라면 알려주시면 감사하겠습니다 >_ㅇ!!

```suggestion
    import(`../../../assets/static/markdown/${detailMdName}.md`)
      .then((res) => {
        fetch(res.default)
          .then((res) => res.text())
          .then((res) => setModalContent(mmd(res)))
          .catch((err) => console.log(err));
      })
      .catch((err) => {
        console.log(err);

        return '';
      });
```